File: Java/src/irrgarten/CombatElement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author jorge
 */
public abstract class CombatElement {
    private float effect;
    private int uses;
    
    CombatElement(float e, int u){
        effect=e;
        uses=u;
    }
    
    protected float produceEffect(){
        if (uses>0){
            uses--;
            return effect;
        } else {
            return 0;
        }
    }
    
    public boolean discard(){
        return Dice.discardElement(uses);
    }
    
    @Override
    public String toString(){
       // return "["+String.valueOf(effect)+","+String.valueOf(uses)+"]";
       return "["+effect+","+uses+"]";
    }
}

File: Ruby/CombatElement.rb
#encoding: UTF-8

module Irrgarten
	class CombatElement
		public
		def initialize(effect,uses)
			@effect=effect
			@uses=uses
		end
		
		def produce_effect()
			if (@uses>0)
				@uses-=1
				return @effect
			else
				return 0
			end
		end
		
		def discard
			Dice.discard_element(@uses)
		end
		
		def to_string
			"["+@effect.to_s+","+@uses.to_s+"]"
		end
		
		private_class_method :new
	end
end
